x-pack/plugins/ml/public/application/explorer/swimlane_pagination.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useCallback, useState } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiContextMenuPanel,
+  EuiPagination,
+  EuiContextMenuItem,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+
+interface SwimLanePaginationProps {
+  fromPage?: number;
+  perPage?: number;
+  cardinality: number;
+  onPaginationChange: (arg: { perPage?: number; fromPage?: number }) => void;
+}
+
+export const SwimLanePagination: FC<SwimLanePaginationProps> = ({
+  cardinality,
+  fromPage = 1,
+  perPage = 10,
```

Comment:
Is it worth storing the default page size in `explorer_constants.ts` ?

Reply:
yes, good point. I've added `EMBEDDABLE_DEFAULT_PER_PAGE` which probably should be generic `SWIM_LANE_DEFAULT_PAGE_SIZE`

Reply:
But the pagination component should not be aware of that. I've changed these props to be required.

Reply:
The constant has been introduced in https://github.com/elastic/kibana/pull/70063/commits/57e4caead008be1b15413893ee47265faaa8e897